### DISCUSSION TITLE
Is keycloak affected by CVE-2022-47966

### DISCUSSION THREAD

**Discussion by L3m0nb4tt3ry** [keycloak/keycloak#21593] [Q&A] [answered]
As per the research blog [here](https://blog.viettelcybersecurity.com/saml-show-stopper/) it was found that xmlsec < 2.2.3 is vulnerable and "this does not only impact a particular product". A lot of scanners are flagging this dependency as vulnerable. Can anyone please confirm whether the keycloak is affected by this vulnerability ?

**Comment by abstractj** [comment, marked as answer]
@L3m0nb4tt3ry no, Keycloak is not vulnerable, nor affected. CVE-2022-47966 indicates that Multiple Zoho ManageEngine, which uses Apache xmlsec version 1.4.1, is affected. However, Keycloak does not have any dependency with Multiple Zoho, and we use [xmlsec version 2.2.3](https://github.com/keycloak/keycloak/blob/main/pom.xml#L115).

**Reply by L3m0nb4tt3ry** [reply]
@abstractj Thank you for the explanation, this makes sense, however the security researcher drew following conclusions
- This does not only affect ManageEngine products.
- How to fix: Update xmlsec library to version 2.3.0 or higher.
and added following line which is confusing 
```
 I don't know if it is belong to Apache XMLSec, Shibboleth OpenSAML or ManageEngine, they have a lengthy discussion.
```
Hence the question.

**Reply by abstractj** [reply]
@L3m0nb4tt3ry CVE-2022-47966 is a vulnerability in multiple Zoho ManageEngine products. It is related to the way that ManageEngine products handle XSLT requests. However, Keycloak is not vulnerable, or affected by this vulnerability. If you identify a vulnerability or a way to exploit Keycloak, please follow the guidelines here: https://github.com/keycloak/security. 

For CVEs identified on third-party dependencies, please create a new GitHub issue here: https://github.com/keycloak/security/issues/new
